swatch-core/src/main/java/org/candlepin/subscriptions/db/HostRepository.java

```diff
@@ -510,6 +510,14 @@ default Uom getDefaultUomForProduct(String productId) {
   Page<Host> getHostsByHypervisor(
       @Param("orgId") String orgId, @Param("hypervisor_id") String hypervisorId, Pageable pageable);
 
+  @Query(
+      "select distinct h1 from Host h1 where "
+          + "h1.orgId = :orgId and "
+          + "h1.hypervisorUuid in (select h2.subscriptionManagerId from Host h2 where "
```

Comment:
Interesting! I didn't know we were storing the hypervisoruuid with the same subscriptionmanagerid. So, a linked tree within the same table.